src/packet.rs

```diff
@@ -53,34 +67,22 @@ impl<W: Write> PacketWriter<W> {
     pub fn end_packet(&mut self) -> io::Result<()> {
         self.maybe_end_packet()
     }
-}
 
-impl<W> PacketWriter<W> {
-    pub fn set_seq(&mut self, seq: u8) {
-        self.seq = seq;
-    }
-}
+    pub fn switch_to_tls(&mut self, config: Arc<ServerConfig>) -> io::Result<()> {
+        assert!(self.remaining() == 0); // otherwise we've read ahead into the TLS handshake and will be in trouble.
```

Comment:
I missed this last time around — this should use `assert_eq!` so that if it fails we learn how much _was_ remaining,

Reply:
Done.